### PR TITLE
ci: fix shellcheck in test-go

### PR DESCRIPTION
**Pull request by Madhu-1**
Fixed shellcheck in test-go script

fixes https://github.com/ceph/ceph-csi/issues/1634

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: scripts/test-go.sh [hunk 1]
```diff
@@ -20,23 +20,23 @@ failed=0
...
 		mkdir -p coverage
 		fn="${GO_COVER_DIR}/${gopackage////-}.html"
 		echo " * generating coverage html: ${fn}"
 		go tool cover -html=cover.out -o "${fn}"
 	fi
 	rm -f cover.out
-	if [[ ${failed} -ne 0 && ${TEST_EXITFIRST} = "yes" ]]; then
-		exit ${failed}
+	if [[ "${failed}" -ne 0 && "${TEST_EXITFIRST}" = "yes" ]]; then
+		exit "${failed}"
```
Comment (Yuggupta27) [inline scripts/test-go.sh:39]: this

File: scripts/test-go.sh [hunk 2]
```diff
@@ -20,23 +20,23 @@ failed=0
...
 	fi
 done
-exit ${failed}
+exit "${failed}"
```
Comment (Yuggupta27) [inline scripts/test-go.sh:42]: and this

**Review by Yuggupta27 — CHANGES_REQUESTED** [review]
Double quotes are required for variables which can either be empty or contain whitespace. To solve the shellcheck issue for the script, just the following changes should be fine.

**Comment by Yuggupta27** [comment]
Also, using quotes more is also a safer approach, so you can keep the changes

**Review by Yuggupta27 — APPROVED** [review]

File: scripts/test-go.sh [hunk 3]
```diff
@@ -20,23 +20,23 @@ failed=0
 for gopackage in ${GOPACKAGES}; do
 	echo "--- testing: ${gopackage} ---"
 	# shellcheck disable=SC2086
-	go test "${GO_TAGS}" -mod=vendor -v ${GOTESTOPTS} "${gopackage}" || ((failed += 1))
+	go test "${GO_TAGS}" -mod=vendor -v "${GOTESTOPTS}" "${gopackage}" || ((failed += 1))
```
[the remark this answers is not in the document]
Comment (nixpanic) [reply scripts/test-go.sh]: I have updated this PR with a fix to use an array for `${GOTESTOPTS}`. Local testing suggests it works...

**Comment by nixpanic** [comment]
Hmm, **ci/centos/containerized-tests** detects this as a `doc change only`... That is not really correct :-1:

**Review by nixpanic — APPROVED** [review]
Approving this, looks good to me (but I also made a minor correction).

**Comment by nixpanic** [comment]
Marking this `[ready-to-merge]` so that other PRs get unblocked.

**Comment by Madhu-1** [comment]
@mergifyio backport release-v3.1

**Pull request merged**